internal/permissions/groups/middleware.go

```diff
@@ -0,0 +1,90 @@
+package userpermissions
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/sirupsen/logrus"
+	"github.com/source-academy/stories-backend/internal/auth"
+	"github.com/source-academy/stories-backend/internal/database"
+	groupenums "github.com/source-academy/stories-backend/internal/enums/groups"
+	apierrors "github.com/source-academy/stories-backend/internal/errors"
+	"github.com/source-academy/stories-backend/model"
+)
+
+type contextKey string
+
+const (
+	GroupKey contextKey = "group_context"
+	RoleKey  contextKey = "role_context"
+)
+
+func InjectUserGroupIntoContext(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		groupIDStr := chi.URLParam(r, "groupID")
+		groupID, err := strconv.Atoi(groupIDStr)
+		if err != nil {
+			logrus.Error(err)
+			apierrors.ServeHTTP(w, r, apierrors.ClientUnprocessableEntityError{
```

Comment:
On second thought, since this is part of the URL path, and not the request body, 404 Not Found might be more appropriate (i.e. "route" not found, as opposed to group not found).

This would be similar to how nginx or other reverse proxies handle their routing as well.

Reply:
I will take care of this, no worries.